sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -390,6 +391,167 @@ abstract class UnixTimeExprMeta[A <: BinaryExpression with TimeZoneAwareExpressi
+        Math.multiplyExact(maxValue.getLong, multiplier)
+      }
+    }
+    withResource(input.min()) { minValue =>
+      if (minValue.isValid) {
+        Math.multiplyExact(minValue.getLong, multiplier)
+      }
+    }
+  }
+
+  protected val convertTo : GpuColumnVector => ColumnVector
+  
+  override def doColumnar(input: GpuColumnVector): ColumnVector = {
+    convertTo(input)
+  }
+}
+
+case class GpuSecondsToTimestamp(child: Expression) extends GpuNumberToTimestampUnaryExpression {
+
+  override def nullable: Boolean = child.dataType match {
+    case _: FloatType | _: DoubleType => true
+    case _ => child.nullable
+  }
+
+  protected lazy val convertTo: GpuColumnVector => ColumnVector = child.dataType match {
+    case LongType =>
+      (input: GpuColumnVector) => {
+        checkLongMultiplicationOverflow(input.getBase, DateTimeConstants.MICROS_PER_SECOND)
+        val mul = withResource(Scalar.fromLong(DateTimeConstants.MICROS_PER_SECOND)) { scalar =>
+          input.getBase.mul(scalar)
+        }
+        withResource(mul) { mul =>
+          mul.asTimestampMicroseconds()
+        }
+      }
+    case DoubleType | FloatType =>
+      (input: GpuColumnVector) => {
+        GpuCast.doCast(input.getBase, input.dataType, TimestampType, false, false, false)
+      }
+    case dt: DecimalType =>
+      (input: GpuColumnVector) => {
+        // SecondsToTimestamp supports decimals with a scale of 6 or less, which can be represented
+        // as microseconds. An exception will be thrown if the scale is more than 6.
+        val decimalTypeSupported = DType.create(DType.DTypeEnum.DECIMAL128, -6)
+        if (dt.scale > 6) {
+          // Match the behavior of `BigDecimal.longValueExact()`, if decimal is equal to the value
+          // casted to decimal128 with scale 6, then no rounding is necessary.
+          val decimalTypeAllScale = DType.create(DType.DTypeEnum.DECIMAL128, -dt.scale)
+          val isEqual = withResource(input.getBase.castTo(decimalTypeSupported)) { casted =>
+            withResource(casted.castTo(decimalTypeAllScale)) { castedAllScale =>
+              withResource(input.getBase.castTo(decimalTypeAllScale)) { original =>
+                original.equalTo(castedAllScale)
+              }
+            }
+          }
+          val roundUnnecessity = withResource(isEqual) { isEqual =>
+            withResource(isEqual.all()) { all =>
+              all.isValid && all.getBoolean()
+            }
+          }
+          if (!roundUnnecessity) {
+            throw new ArithmeticException("Rounding necessary")
+          }
+        }
+        // Cast to decimal128 to avoid overflow
+        val mul = withResource(input.getBase.castTo(decimalTypeSupported)) { decimal =>
+          withResource(Scalar.fromLong(DateTimeConstants.MICROS_PER_SECOND)) { scalar =>
+            decimal.mul(scalar, decimalTypeSupported)
+          }
+        }
+        // Match the behavior of `BigDecimal.longValueExact()`:
```

Comment:
You need to protect the `mul` when computing the `largerThanLongMax` and `smallerThanLongMin`. You can do it like
```
closeOnExcept(mul) { _ =>
   val greaterThanMax = ....
   val smallerThanLongMin = ....
   if (largerThanLongMax || smallerThanLongMin) {
       throw new java.lang.ArithmeticException("Overflow")
    }
}
```